pkg/k8sclient/k8sclient.go

```diff
@@ -3,26 +3,76 @@ package k8sclient
 import (
 	"errors"
 	"context"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"github.com/sbabiv/roundrobin"
 )
 
 const (
-	NodeSelector = "nokia.k8s.io/nodeSelector"
-	NodeName = "nokia.k8s.io/nodeName"
-	LvCapacity = "nokia.k8s.io/lv-capacity"
+	LvCapacity 				 = "nokia.k8s.io/lv-capacity"
 	LocalScProvisioner = "nokia.k8s.io/local"
+	NodeName           = "nokia.k8s.io/nodeName"
+	RR								 = "round robin"
+	Cap								 = "capacity"
+)
+
+var (
+	rr = roundrobin.Balancer{}
```

Comment:
you absolutely did not eliminate package level globals. I also don't understand why did you refactor the package interface from a better design to a worse one

the package shall create a K8s Client object **on demand**, and return it to the caller. the object shall already include an RR member, which is **automatically initialized** during the creation of the parent object (RAII).

the client methods shall take this object as their input parameter, and use the K8s client and the RR object members